internal/server/server.go

```diff
@@ -62,8 +72,33 @@ func NewServer(repo *repository.Repository, tokenManager *auth.TokenManager) *Se
 // AuthMiddleware checks user authorization.
 func (s *Server) AuthMiddleware(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// check JWT tokens
-		next.ServeHTTP(w, r)
+		authHeader := r.Header.Get(AuthorizationHeader)
+		if authHeader == "" {
+			http.Error(w, "empty auth handler", http.StatusUnauthorized)
+			return
+		}
+
+		authHeaderParts := strings.Split(authHeader, " ")
+		if len(authHeaderParts) != 2 || authHeaderParts[0] != NeededSecurityScheme {
+			http.Error(w, "invalid auth header", http.StatusUnauthorized)
+			return
+		}
+
+		if len(authHeaderParts[1]) == 0 {
+			http.Error(w, "token is empty", http.StatusUnauthorized)
+			return
+		}
+
+		token := authHeaderParts[1]
+		claimsUserID, err := s.tokenManager.ParseToken(token)
+		if err != nil {
+			http.Error(w, "can't parse JWT", http.StatusUnauthorized)
+			return
+		}
+
+		ctx := context.WithValue(r.Context(), userIDKey, claimsUserID)
```

Comment:
Neat! Let's create two functions for this. To put login into context and to retrieve login from it. They'll use `userIDKey` to do it.

Reply:
`func contextWithLogin(ctx context.Context, login string) context.Context`
`func LoginFromContext(ctx context.Context) string`

Reply:
> `func contextWithLogin(ctx context.Context, login string) context.Context`
> `func LoginFromContext(ctx context.Context) string`

OK, but here only user id is used